core/retries/src/main/java/software/amazon/awssdk/retries/internal/circuitbreaker/TokenBucket.java

```diff
@@ -89,8 +89,7 @@ public ReleaseResponse release(int amountToRelease) {
                            .maxCapacity(maxCapacity);
 
         if (amountToRelease == 0) {
-            return builder.currentCapacity(capacity.get())
-                          .build();
+            amountToRelease = 1;
```

Comment:
Why are we releasing zero? Should the fix be in the caller instead of here?

Reply:
Could be the the caller, no particular reason. We release zero on a successful first attempt since we have not withdrawn any tokens.

Reply:
Imo, doing something different from what's asked in the token bucket is probably not the move. Calling release(0) and it transforming to release(1) is confusing.

Reply:
Calling release(1) on success even without removing anything is fine. It's not a zero-sum game. The token bucket enforces a max on capacity, and we already remove more than we add for a given request, so adding more than we remove is a fine action as well.

Reply:
We don't know at success if we withdraw or not, so it needs to be something along the lines of 

```
release(Math.max(1, withdrawn))
```

Let me know if that looks better.

Reply:
I pushed a commit with the alternative.